nixos/modules/services/search/quickwit.nix

```diff
@@ -0,0 +1,74 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.quickwit;
+  settingsFormat = pkgs.formats.yaml { };
+  configFile = settingsFormat.generate "quickwit.yaml" cfg.settings;
+in
+{
+  options.services.quickwit = {
+    package = mkOption {
+      type = types.package;
+      description = "Quickwit package to use.";
+    };
+
+    enable = mkEnableOption "Quickwit search engine";
+
+    environmentFile = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      description = "Environment file to pass to the systemd service.";
+    };
+
+    settings = mkOption {
+      description = "Quickwit settings.";
+      type = types.submodule {
+        freeformType = settingsFormat.type;
+
+        options = {
+          listen_address = mkOption {
+            type = types.str;
+            default = "127.0.0.1";
+            description = "Address to listen on.";
+          };
+
+          data_dir = mkOption {
+            type = types.path;
+            default = "/var/lib/quickwit";
+            description = "Path to the directory where Quickwit will store its data.";
+          };
+
+        };
+      };
+    };
+  };
+  config = mkIf cfg.enable {
+    users.users.quickwit = {
+      description = "Quickwit daemon user";
+      isSystemUser = true;
+      group = "quickwit";
+    };
+    users.groups.quickwit = {};
+    system.activationScripts.makeQuickwitDir = ''
+      mkdir -p ${cfg.settings.data_dir}
+      chown -R quickwit:quickwit ${cfg.settings.data_dir}
```

Comment:
This works. 
I'm testing the full setup each time (with nixos-vm).

Reply:
@tcheronneau The fact that it works for you doesn't make it correct.

The path needs quoting because it is incorrect otherwise.

If you set (do not try this) `data_dir = "/ my folder containing a space";`, the above will delete your computer due to lack of quoting.